doc/research/nextcloud_integration.md

```diff
@@ -9,45 +11,208 @@
 
-### testing with cadaver
+Now we can make requests against the '/remote.php/webdav' endpoint.
+This endpoint enables us to fetch files starting from the root directory of the logged in user.
+Given the previously defined directory structure we can now access the file 'secret-doc.txt' with the following URI: 'https://cloud.permaplant.net/remote.php/webdav/Documents/secret-doc.txt'
 
-opening a connection
+### Directory Structure
 
+In the PermaplanT Nextcloud instance we have following directory structure for each user:
 ```bash
-cadaver
-open https://nextcloud.markus-raab.org/nextcloud/remote.php/dav/files/USERNAME/Documents
+PermaplanT/
+└── Maps
+    ├── map_01
+    │   ├── BaseLayer
+    │   │   └── base_layer_image.png
+    │   └── PhotoLayer
+    │       ├── photo1.jpg
+    │       └── photo2.jpg
+    └── map_02
+        ├── BaseLayer
+        │   └── base_layer_image.png
+        └── PhotoLayer
+            ├── photo1.jpg
+            └── photo2.jpg
 ```
 
-type help for a list of commands
+When a map is shared between different users the directory also has to be shared and placed within this structure.
+Therefore the map directories must have a globally unique name.
+The name of each map directory is the same as the name of the map in the PermaplanT database. This is possible because the name in the database is unique.
+
+All public directories can be placed anywhere as they are identified by the public share token.
+They are not visible to all of the users in Nextcloud, so they are not included in the directory structure above.
+
+### Sharing a Map
+
+When we want to add additional members to our PermaplanT map we also have to share the Nextcloud resources with them.
+First we need to create a circle.
+This can be achieved by following API call:
+```json
+"method": "POST",
+"headers": [
+  "OCS-APIRequest": true,
+  "Content-Type": "application/json"
+],
+"body": {
+  "name": "map_01",
+  "personal": false,
+  "local": false
+},
+"scheme":	"https",
+"host": "cloud.permaplant.net",
+"filename": "/ocs/v2.php/apps/circles/circles"
+```
+The name of the circle is the same as the name of the map.
+Unfortunately the circle API is not documented. The implementation of the API can be found on the corresponding [github repository](https://github.com/nextcloud/circles/blob/22238597fb9045e748119247fceaac7321f0a31e/appinfo/routes.php).
 
-```bash
-help
+Now the circle has to be added to the shares for the map directory.
+This can be done with the share API which is documented [here](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-share-api.html).
+Following API call can be used:
 ```
+"method": "post",
+"headers": [
+  "ocs-apirequest": true
+],
+"body": {
+  "path": "PermaplanT/Maps/map_01",
+  "shareType": 7,
+  "shareWith": "<circleId>"
+},
+"scheme":	"https",
+"host": "cloud.permaplant.net",
+"filename": "/ocs/v2.php/apps/files_sharing/api/v1"
+```
+
+### Components
```

Comment:
Probably better to document them directly in Storybook, isn't it?

```suggestion
### React Components
```

Reply:
Yes, they should be in Storybook as well.